extract-files.sh

```diff
@@ -0,0 +1,187 @@
+#!/bin/bash
+
+# Copyright (C) 2010 The Android Open Source Project
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+if [ -z "${DEVICE}" ]; then
+    echo "You need to define the DEVICE."
+    exit 1
+fi;
+
+if [ -z "${MANUFACTURER}" ]; then
+    echo "You need to define the MANUFACTURER."
+    exit 1
+fi;
+
+if [[ -z "${ANDROIDFS_DIR}" ]]; then
+    ANDROIDFS_DIR=../../../backup-${DEVICE}
+fi
+
+if [[ ! -d ${ANDROIDFS_DIR}/system/etc/firmware/modem.mdt ]]; then
+    echo Backing up necessary files to backup-${DEVICE}
+    adb root &&
+    sleep 1 &&
+    adb wait-for-device &&
+    adb pull /system/vendor/lib/libbt-vendor.so ${ANDROIDFS_DIR}/system/vendor/lib/libbt-vendor.so &&
+    adb pull /system/etc/firmware ${ANDROIDFS_DIR}/system/etc/firmware
+fi
+
+echo Pulling files from ${ANDROIDFS_DIR}
+
+# Always true, before there is a reference base image.
+DEVICE_BUILD_ID=23.0.1.A.5.77
```

Comment:
OK, I'll update it to 23.1.A.1.28, but I didn't the update or image of 23.1.A.1.28.